tools/client/workloads.go

```diff
@@ -19,8 +19,14 @@ func (w *httpWorkloads) Create(reservation workloads.Reservation) (id schema.ID,
 	return
 }
 
-func (w *httpWorkloads) List(page *Pager) (reservation []workloads.Reservation, err error) {
+func (w *httpWorkloads) List(nextAction workloads.NextActionEnum, customerTid int64, page *Pager) (reservation []workloads.Reservation, err error) {
 	query := url.Values{}
+	if nextAction != 0 {
```

Comment:
Actually nextAction of value 0 is a valid value. (check available values for the enum)
To work around this u can either do
- (check if value is < 0) and then caller has to pass of `-1` to donate (not set) (I don't like this very much)
- Or, you can change the input to `*NextActionEnum` then a nil value is possible, and then u only pass pointer to the value if u want to filter with next action. then u can do `if nextAction != nil` then `query.Set("next_action", *nextAction)`

Reply:
I just realized that the first option is not even possible because the Enum type is of size `uint8` so no support for minus value. Hence only possible option is using pointer value